src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
+            FS.writeFile(path, new Uint8Array(data), { encoding: 'binary' });
```

Comment:
It is a good question, I'm not sure. Both approaches have their merits.

If emscripten automatically deletes the files after the callback, then possible memory leaks would be avoided. However, this would require the application finish using the app after the `glfwSetDropCallback` callback returns, plausible for simpler apps (I think mine will cope with it just fine), but it is not a limitation documented by the glfw API that the file cannot be used after the callback returns. Native apps reasonably would expect the file still to be usable, perhaps performing some multi-threaded processing afterwards, though I don't know how common this is.

Requiring the app to unlink the file itself would introduce the need for `#ifdef __EMSCRIPTEN__ unlink(file)` to fix possible memory leaks, but on the other hand would allow apps which depend on the file existing after the call to function. This seems safer and more compatible, leaning towards this second option.